app/models/comment.rb

```diff
@@ -98,9 +113,22 @@ class Comment < AbstractModel
   after_create :notify_users
   after_create :oil_and_water
 
+  scope :by_user,
+        ->(user) { where(user: user) }
+  scope :for_user,
+        lambda { |user|
+          targets = []
+          all_types.each do |model|
+            targets |= model.where(user: user).map(&:id)
+          end
+          where(target: targets)
+        }
+  scope :for_target,
+        ->(target) { where(target: target) }
+
   # Returns Array of all models (Classes) which take comments.
   def self.all_types
-    [Location, Name, Observation, Project, SpeciesList]
+    all_type_tags.map { |t| t.to_s.camelize.constantize }
   end
```

Comment:
Is this unnecessary DRY? Does it force a class lookup on all those type tags? Yes, it does...

The inverse, deriving the tags from a list of classes,  would also reference the classes.
Maybe best to leave it like it was, a double list.